homeassistant/helpers/selector.py

```diff
@@ -1080,10 +1081,15 @@ def __init__(self, config: NumberSelectorConfig | None = None) -> None:
         """Instantiate a selector."""
         super().__init__(config)
 
-    def __call__(self, data: Any) -> float:
+    def __call__(self, data: Any) -> float | int:
         """Validate the passed selection."""
         value: float = vol.Coerce(float)(data)
 
+        if (step := self.config["step"]) != "any" and step.is_integer() and step > 0:
```

Comment:
The step size could be `1.5`

Reply:
My idea is that:
- step size 1 would trigger integer validation
- step size 1.5 would be happy with float
- step size 2 would trigger integer validation